packages/config/src/VitessceAutoConfig.js

```diff
@@ -185,55 +185,91 @@ class AnndataZarrAutoConfig extends AbstractAutoConfig {
     return views;
   }
 
-  async setMetadataSummary() {
-    if (Object.keys(this.metadataSummary).length > 0) {
-      return this.metadataSummary;
+  async setMetadataSummaryWithZmetadata(response) { /* eslint-disable-line class-methods-use-this */
+    const metadataFile = await response.json();
+    if (!metadataFile.metadata) {
+      throw new Error('Could not generate config: .zmetadata file is not valid.');
     }
 
-    const parseMetadataFile = (metadataFile) => {
-      if (!metadataFile.metadata) {
-        throw new Error('Could not generate config: .zmetadata file is not valid.');
-      }
+    const obsmKeys = Object.keys(metadataFile.metadata)
+      .filter(key => key.startsWith('obsm/X_'))
+      .map(key => key.split('/.zarray')[0]);
 
-      const obsmKeys = Object.keys(metadataFile.metadata)
-        .filter(key => key.startsWith('obsm/X_'))
-        .map(key => key.split('/.zarray')[0]);
+    const obsKeysArr = Object.keys(metadataFile.metadata)
+      .filter(key => key.startsWith('obs/')).map(key => key.split('/.za')[0]);
 
-      const obsKeysArr = Object.keys(metadataFile.metadata)
-        .filter(key => key.startsWith('obs/')).map(key => key.split('/.za')[0]);
+    function uniq(a) {
+      return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
+    }
+    const obsKeys = uniq(obsKeysArr);
 
-      function uniq(a) {
-        return a.sort().filter((item, pos, ary) => !pos || item !== ary[pos - 1]);
-      }
-      const obsKeys = uniq(obsKeysArr);
+    const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+
+    return {
+      obsm: obsmKeys,
+      obs: obsKeys,
+      X: X.length > 0,
+    };
+  }
 
-      const X = Object.keys(metadataFile.metadata).filter(key => key.startsWith('X'));
+  async setMetadataSummaryWithoutZmetadata() {
+    const suffixes = [
+      '/obsm/X_pca/.zarray',
+      '/obsm/X_umap/.zarray',
+      '/obsm/X_tsne/.zarray',
+      '/obsm/X_spatial/.zarray',
+      '/obsm/X_segmentations/.zarray',
+      '/obs/.zattrs',
+      '/X/.zarray',
+      '/X/data/.zarray',
+    ];
 
-      const out = {
-        obsm: obsmKeys,
-        obs: obsKeys,
-        X: X.length > 0,
-      };
+    const promises = suffixes.map(suffix => fetch(this.fileUrl + suffix));
 
-      return out;
+    const results = await Promise.all(promises);
+    const metadataSummary = {
+      obsm: [],
+      obs: [],
+      X: false,
     };
 
+    const obsJson = results.find(j => j.ok && j.url.startsWith(`${this.fileUrl}/obs/.zattrs`));
+
+    if (obsJson) {
+      const obsColumns = await obsJson.json();
+      obsColumns['column-order'].forEach(key => metadataSummary.obs.push(`obs/${key}`));
+    }
+
+    results
+      .filter(j => j.ok)
+      .forEach((j) => {
+        if (j.url.startsWith(`${this.fileUrl}/obsm`)) {
+          metadataSummary.obsm.push(
+            j.url.replace(this.fileUrl, '').replace('/', '').replace('/.zarray', ''),
```

Comment:
What is this line doing - can you add a comment? Also, can we use substring and the length of `this.fileUrl` instead of `.replace`?

Reply:
This line is stripping everything from `j.url`, leaving only `obsm/` and what is immediately after it (for example, `X_segmentations`, or `X_spatial`).

Reply:
It is not very readable, even with adding a comment, I decided it is best to refactor it a little and write a getter function for this.